tooling/reorgs/src/main.rs

```diff
@@ -76,6 +87,85 @@ where
     info!("");
 }
 
+async fn no_reorgs_full_sync_smoke_test(simulator: Arc<Mutex<Simulator>>) {
+    let mut simulator = simulator.lock().await;
+
+    // Start two ethrex nodes
+    let node0 = simulator.start_node().await;
+    let node1 = simulator.start_node().await;
+
+    // Create a chain and extend it with a few empty blocks
+    let base_chain = node0.extend_chain(simulator.get_base_chain(), 10).await;
+
+    // Try to fully sync node1 (which is a peer of node0)
+    node1.update_forkchoice(&base_chain).await;
+}
+
+async fn test_reorg_back_to_base(simulator: Arc<Mutex<Simulator>>) {
+    let mut simulator = simulator.lock().await;
+
+    // Start two ethrex nodes
+    let node0 = simulator.start_node().await; // Test_node
+    let node1 = simulator.start_node().await;
+
+    let base_chain = simulator.get_base_chain();
+    let side_chain = base_chain.fork();
+    // Create a chain and extend it with a few empty blocks
+    let base_chain = node1.extend_chain(base_chain, 10).await;
+
+    // Create another chain (a fork of the first one) and extend it with a few empty blocks
+    let _ = node0.extend_chain(side_chain, 10).await;
+
+    // Try to fully sync node0 to base chain
+    node0.update_forkchoice(&base_chain).await;
+}
+
+async fn test_reorg_back_to_base_with_common_ancestor(simulator: Arc<Mutex<Simulator>>) {
+    let mut simulator = simulator.lock().await;
+
+    // Start two ethrex nodes
+    let node0 = simulator.start_node().await; // Test_node
+    let node1 = simulator.start_node().await;
+
+    let base_chain = simulator.get_base_chain();
+    // Create a chain and extend it with a few empty blocks
+    let base_chain = node1.extend_chain(base_chain, 10).await;
+
+    // Update node0 to the base chain
+    node0.update_forkchoice(&base_chain).await;
+
+    let side_chain = base_chain.fork();
+    // Extend the base chain
+    let base_chain = node1.extend_chain(base_chain, 10).await;
+    // Create another chain (a fork of the first one) and extend it with a few empty blocks
+    let _ = node0.extend_chain(side_chain, 10).await;
+
+    // Try to fully sync node0 to base chain
+    node0.update_forkchoice(&base_chain).await;
+}
+
+async fn reorgs_full_sync_smoke_test(simulator: Arc<Mutex<Simulator>>) {
```

Comment:
Let's rename it to `test_chain_split`, since that's the specific scenario we're testing.